### ISSUE TITLE
Make sure thread composer selects correct defaults in all instances

### ISSUE THREAD

**Issue by mxstbr** [withspectrum/spectrum#4724]
Apparently it doesn't on the channel view: https://spectrum.chat/users/josh-padnick?thread=ef588dc2-ae37-4755-9b2b-25615e67dd78

I think we should not select anything by default to avoid people posting to the wrong community. You should always have to select a community and channel explicitly, unless you are on a community view (community gets defaulted) or a channel view (both get defaulted).

I think we should maybe hide the selects entirely on the channel view, and hide the community selector on the community view. It's strange to be able to post to a different community from another community.

**Comment by brianlovin** [comment]
Not a bad idea at all to make it entirely user-generated to avoid so many people accidentally posting in the wrong channel.

**Comment by mxstbr** [comment]
We also need to make sure to set the defaults correctly in the inbox view with community/channel filters, we currently do not at all!!! https://spectrum.chat/spectrum/hugs-n-bugs/clicking-on-create-new-conversation-changes-the-channel~ae0d34ba-2a06-4b37-910d-05d453279700

That might be why so many people are posting in the wrong communities.

**Comment by brianlovin** [comment]
Spent an hour and a half trying to think of how to clean up our composer spaghetti...this is really painful. For documentation, here's how the behavior should probably work:

1. If the user is either viewing a community or has a selected community in the inbox, that community should be "locked" as the community the composer is posting to. The user will only be able to change channels within that locked community. The channel selection should be empty by default; users must choose which channel to post to.
2. If the user is either viewing a channel or has a selected channel in the inbox, that channel and community should both be "locked" as the community and channel the composer is posting to. The user will not be able to change the community or channel selection.
3. If the user is viewing `/new/thread` or is viewing the inbox "everything" feed, the user won't have any locked selection. Instead, they should see a prompt to choose which community to post their thread to. Once they've chosen the community, a second dropdown should fetch that community's channels and prompt the user to select a channel. In this way the user is opting in to *both* community and channel selection. We will no longer default to anything in this situation.

Basically I'm not really sure where to store this state, if it should be stored at all or calculated on the fly every time the composer is opened. One route I started exploring was storing all these locked defaults in redux state and dispatching selections when the user viewed a community/channel or selected one in the inbox. This feels too...messy.

Any quick high-level approach ideas here @mxstbr ?

**Comment by mxstbr** [comment]
Thanks for digging into this and the detailed notes!

----

First idea that jumps to mind is making this URL-driven? Clicking on the "New post" icon redirects to `/new/thread` on mobile (full-screen view; existing behavior), and on desktop it adds `?new` to the existing URL (slider view).

Then we can add more params like `?new&community=spectrum&channel=general` (`/new/thread?community=spectrum&channel=general`) to the links across the board. Depending on these params being both being set, just the community being set or none being set we lock that selection in place and hide/disable the select.

Could that work?

**Comment by brianlovin** [comment]
Url driven is probably a good idea, yeah - will have to make it play nicely with `?t=` in the inbox, but otherwise this is a good idea!

Will give it a try this afternoon.

**Comment by brianlovin** [comment]
Just realized the url behavior will also fix the selection problems on mobile!
